### PR TITLE
[SPARK-34022][DOCS] Support latest mkdocs in SQL built-in function docs

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR adds the support of the latest mkdocs, and makes the sidebar properly show. It works in lower versions too.

Before:

Screen Shot 2021-01-06 at 5 11 56 PM

After:

Screen Shot 2021-01-06 at 5 10 53 PM

### Why are the changes needed?

This is a regression in the documentation.

### Does this PR introduce _any_ user-facing change?

Technically no. It's not related yet. It fixes the list on the sidebar appears properly.

### How was this patch tested?

Manually built the docs via `./sql/create-docs.sh` and `open ./sql/site/index.html`

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @maropu FYI

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. It looks nice. :)

**Comment by HyukjinKwon** [comment]
Thank you @dongjoon-hyun!

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

**Comment by maropu** [comment]
Oh, I see. Nice update. Late lgtm.

File: sql/gen-sql-api-docs.py [hunk 1]
```diff
@@ -195,6 +195,7 @@ def generate_sql_api_markdown(jvm, path):
     """
 
     with open(path, 'w') as mdfile:
+        mdfile.write("# Built-in Finctions\n\n")
```
Comment (jaceklaskowski) [inline sql/gen-sql-api-docs.py:198]: A typo in `Finctions`

**Pull request closed without merging**